tests/BaseAppTest.php

```diff
@@ -50,14 +53,14 @@ public function testBoot()
         $this->app->register(new TestBootableServiceProvider());
         $this->app->boot();
 
-        $this->assertTrue($this->app->get('testRegister'));
-        $this->assertTrue($this->app->get('testBoot'));
+        $this->assertTrue($this->app->get('testRegister')->registered);
+        $this->assertTrue($this->app->get('testBoot')->booted);
```

Comment:
How about registering a mock service provider instead of `TestBootableServiceProvider ` and set expectations for register and boot?